frontend/demo/component/badge/badge-interactive.ts

```diff
@@ -1,22 +1,108 @@
-import 'Frontend/demo/init'; // hidden-full-source-line
+import { getPeople } from 'Frontend/demo/domain/DataService'; // hidden-full-source-line
+import 'Frontend/demo/init';
 
-import { html, LitElement, customElement } from 'lit-element';
-import '@vaadin/vaadin-checkbox/vaadin-checkbox';
+import '@vaadin/vaadin-button';
+import '@vaadin/vaadin-combo-box';
+import '@vaadin/vaadin-icons/vaadin-icons';
+import '@vaadin/vaadin-lumo-styles/icons';
+import '@vaadin/vaadin-ordered-layout';
+import type { ButtonElement } from '@vaadin/vaadin-button';
+import type { ComboBoxElement } from '@vaadin/vaadin-combo-box';
+import { html, LitElement, customElement, internalProperty, css } from 'lit-element';
 import { applyTheme } from 'Frontend/generated/theme';
+import { repeat } from 'lit-html/directives/repeat';
+
+type Profession = string;
 
 @customElement('badge-interactive')
 export class Example extends LitElement {
+  static get styles() {
+    return css`
+      .professions {
+        display: flex;
+        flex-wrap: wrap;
+      }
+
+      .professions > span {
+        margin-inline-end: var(--lumo-space-m);
+        margin-block-end: var(--lumo-space-m);
+      }
+
+      .professions vaadin-button {
+        margin-inline-start: var(--lumo-space-s);
+      }
+    `;
+  }
+
+  @internalProperty()
+  private items: readonly Profession[] = [];
+
+  @internalProperty()
+  private selectedProfessions: readonly Profession[] = [];
+
   constructor() {
     super();
     // Apply custom theme (only supported if your app uses one)
     applyTheme(this.shadowRoot);
   }
 
+  async firstUpdated() {
+    const { people } = await getPeople();
+    this.items = [...new Set(people.map(({ profession }) => profession))];
+  }
+
   render() {
+    // tag::snippet[]
     return html`
-      <!-- tag::snippet[] -->
-      <span theme="badge pill contrast">Badge</span>
-      <!-- end::snippet[] -->
+      <vaadin-vertical-layout theme="spacing">
+        <vaadin-combo-box
+          label="Profession"
+          .items="${this.items}"
+          @change="${this.onChange}"
+        ></vaadin-combo-box>
+        <div class="professions">
+          ${repeat(
+            this.selectedProfessions,
+            (profession) => profession,
+            (profession) => html`
+              <span theme="badge contrast">
```

Comment:
Should be using pill style as in design doc.
```suggestion
              <span theme="badge pill contrast">
```